utils/python-rpc/decoy_selection.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python3
+
+''' See step-by-step accompaniment in docs/DECOY_SELECTION.md '''
+
+import argparse
+import bisect
+try:
+    import numpy as np
```

Comment:
Why require numpy? It's an incredibly large dependency for a very small set of features which the built-in random module also provides.